eZ/Publish/Core/Repository/Helper/DomainMapper.php

```diff
@@ -805,4 +838,16 @@ public function getUniqueHash($object)
     {
         return md5(uniqid(get_class($object), true));
     }
+
+    /**
+     * Returns true if given location is a tree root.
+     *
+     * @param \eZ\Publish\SPI\Persistence\Content\Location $spiLocation
+     *
+     * @return bool
+     */
+    private function isRootLocation(SPILocation $spiLocation): bool
+    {
+        return self::ROOT_LOCATION_ID === (int)$spiLocation->id;
```

Comment:
AFAIK:
```suggestion
        return $spiLocation->id === $spiLocation->parentId;
```
I also don't like this cast here. If `$spiLocation->id` is sometimes `int` and sometimes `string` it needs to be fixed on a SPI row mapper level.

You could of course cast both values, but this would mean true for `null` values which might be risky ;)

Reply:
Fixed in https://github.com/ezsystems/ezpublish-kernel/pull/2560/commits/8cdb691a43d601f4239526d8628e23ea47a35512